reconcile/utils/pagerduty_api.py

```diff
@@ -1,40 +1,77 @@
-import datetime
+from datetime import datetime as dt
+import logging
+from typing import Callable
+from typing import Iterable
+from typing import Optional
+from typing import Protocol
 
-import requests
 import pypd
+import requests
+from pydantic import BaseModel
+from sretoolbox.utils import retry
 
 from reconcile.utils.secret_reader import SecretReader
+from reconcile.utils.secret_reader import SupportsSecret
+
+
+class PagerDutyTargetException(Exception):
+    ...
+
+
+class SupportsPagerDutyInstance(Protocol):
+    name: str
+
+    @property
+    def token(self) -> SupportsSecret:
+        ...
+
+
+class SupportsUser(Protocol):
```

Comment:
naming nit: `PagerDutyUser` instead of `SupportsUser`
same for other `Supports*` Protocols. the `Supports` prefix throws me off a bit

Reply:
I'm following just the patterns, e.g. https://github.com/app-sre/qontract-reconcile/blob/master/reconcile/utils/secret_reader.py#L19. Looks like it's kind of a [standard](https://mypy.readthedocs.io/en/stable/protocols.html#simple-user-defined-protocols).

Reply:
afaiu `Supports*` is the standard if an object supports specific functionality. in the sense of `this objects supports to be closed`.

`SupportsUser` in the sense of `this objects supports BEING a user` sounds awkward to me. that might be just personal preference.

besides the `Supports*` and `*able` Protocols, there are others as well in core python, e.g. `Container` or `Sized`. i think i read somewhere that such names are preferable if a protocol expresses a clear concept of an object being a thing.

i admit, that this PR is not the right context to discuss this. so feel free to ignore.

Reply:
https://github.com/python/typeshed/pull/4229/files

Reply:
[a616877](https://github.com/app-sre/qontract-reconcile/pull/3002/commits/a616877dc1cf01fdcbfadaa4384d87c8fb3952be) ❤️

Reply:
I won't die on that hill :) I've renamed the classes. To be honest, I like it without `Supports`

Reply:
I definitely think this deserves a reference in our style guide. FWIW, I also prefer the new names ;-)